include/deal.II/distributed/cell_data_transfer.h

```diff
@@ -0,0 +1,346 @@
+     * serialized and deserialized in the same order.
+     *
+     * For serialization, the following code snippet saves not only the
+     * triangulation itself, but also the cell by cell data attached:
+     * @code
+     * Vector<double> data_to_transfer (triangulation.n_active_cells());
+     * //[fill data_to_transfer with cell by cell values...]
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * cell_data_trans.prepare_serialization(data_to_transfer);
+     *
+     * triangulation.save(filename);
+     * @endcode
+     *
+     * Later, during deserialization, both triangulation and data can be
+     * restored as follows:
+     * @code
+     * //[create coarse mesh...]
+     * triangulation.load(filename);
+     *
+     * parallel::distributed::CellDataTransfer<dim, spacedim, Vector<double>>
+     *   cell_data_trans(triangulation);
+     * Vector<double> transferred_data (triangulation.n_active_cells());
+     * cell_data_trans.deserialize(transferred_data);
+     * @endcode
+     *
+     * @note If you use more than one object to attach data to a
+     * parallel::distributed::Triangulation at the same time (e.g. a
+     * parallel::distributed::SolutionTransfer object) with the aim of
+     * serialization, the calls to
+     * parallel::distributed::CellDataTransfer::prepare_serialization()
+     * and parallel::distributed::SolutionTransfer::prepare_serialization(),
+     * as well as parallel::distributed::CellDataTransfer::deserialize() and
+     * parallel::distributed::SolutionTransfer::deserialize(),
+     * need to be in the same order, respectively.
+     *
+     * @note See documentation of parallel::distributed::SolutionTransfer for
+     * matching code snippets for both transfer and serialization.
+     *
+     * @ingroup distributed
+     * @author Marc Fehling, 2018
+     */
+    template <int dim, int spacedim = dim, typename VectorType = Vector<double>>
+    class CellDataTransfer
+    {
+    public:
+      /**
+       * Whenever data shall be transferred across coarsening, it is not a
+       * trivial choice how to process data of cells which will be coarsened. Or
+       * in other words, which data should be stored in the corresponding parent
+       * cell.
+       *
+       * In this namespace, we offer a few strategies that cope with this
+       * problem, that can be passed to the
+       * parallel::distributed::CellDataTransfer constructor.
+       */
+      struct CoarseningStrategies
+      {
+        /**
+         * Evaluate data from @p input_vector on children of @p parent.
+         * Check if data on all children match, and return that value.
+         */
+        static typename VectorType::value_type
+        check_equality(const typename parallel::distributed::
+                         Triangulation<dim, spacedim>::cell_iterator &parent,
+                       const VectorType *input_vector)
```

Comment:
can you adjust the interface so that a reference instead of a pointer is passed here? same for the other functions of course